### ISSUE TITLE
GRDB is mixing up order of statement arguments under some circumstances

### ISSUE THREAD

**Issue by mallman** [groue/GRDB.swift#758]
### What did you do?

I wrote a SQL update query using the record API with two statement arguments. I noticed that GRDB was switching the order of the statement argument values: it was setting the value of the first argument to the value of the second argument and vice-versa. I think the circumstances under which this occurs is in updates from joins. Here's a reproducible example:

#### SQL

Create `/tmp/test.db` with

```sql
create table address(id int primary key, isResidential boolean);
create table person(id int primary key, rank int, addressId int references address);

insert into address values (0, 0);
insert into person values (0, 0, 0);
```

#### Swift

```swift
import GRDBCustomSQLite // I'm assuming the fact that I'm using
                        // GRDBCustomSQLite instead of GRDB is irrelevant

struct Person: Codable, PersistableRecord {
  static var databaseTableName = "person"

  let id: Int
  let rank: Int
  let addressId: Int
}

struct Address: Codable, PersistableRecord {
  static var databaseTableName = "address"

  let id: Int
  let isResidential: Bool

  static let people = hasMany(Person.self)
  var people: QueryInterfaceRequest<Person> {
    return request(for: Address.people)
  }
}

let testDbPath = "/tmp/test.db"
var queueConfig = Configuration()
queueConfig.trace = { print($0) }
let dbQueue = try DatabaseQueue(path: testDbPath, configuration: queueConfig)

let personAlias = TableAlias()
let request = Address
  .joining(required: Address.people.aliased(personAlias))
  .filter(personAlias[Column("rank")] >= 10)

try dbQueue.write { db in
  try request.updateAll(db, Column("isResidential") <- 1)
}
```

### What did you expect to happen?

```sql
UPDATE "address" SET "isResidential" = 1 WHERE rowid IN (
  SELECT "address"."rowid" FROM "address"
  JOIN "person" ON "person"."addressId" = "address"."id"
  WHERE "person"."rank" >= 10)
```

### What happened instead?

```sql
UPDATE "address" SET "isResidential" = 10 WHERE rowid IN (
  SELECT "address"."rowid" FROM "address"
  JOIN "person" ON "person"."addressId" = "address"."id"
  WHERE "person"."rank" >= 1)
```

### Environment

**GRDB flavor(s):** GRDBCustomSQLite
**GRDB version:** 4.12.2
**Installation method:** manual Xcode build
**Xcode version:** 11.4 (11E146)
**Swift version:** 5.2
**Platform(s) running GRDB:** macOS
**macOS version running Xcode:** 10.15.4 (19E287)

**Comment by groue** [comment]
Hello @mallman,

Thanks for the reproducible case. Do you have a workaround until a fix is shipped?

**Comment by mallman** [comment]
> Thanks for the reproducible case. Do you have a workaround until a fix is shipped?

I do not, however I have plenty else to work on. This will not keep me from making progress on my project. Thank you, as always, for your terrific work.

**Comment by groue** [comment]
The fix will ship in v4.13.0, shortly. I'll also push the deprecation of the `<-` operator, which was ill-advised. This deprecation will ease the transition to GRDB 5. You are paying the price for a rushed feature, unfortunately!

**Comment by mallman** [comment]
I had a feeling you'd make short work of this. (Where's the ninja emoji when I need one???)

**Comment by groue** [comment]
Believe me, it's much easier when wild users like you embrace the api and make the most of it :-)

**Comment by groue** [comment]
Fixed in [v4.13.0](https://github.com/groue/GRDB.swift/tree/v4.13.0)
